soh/soh/Enhancements/resolution-editor/ResolutionEditor.cpp

```diff
@@ -0,0 +1,421 @@
+                    if (ImGui::Button("Click to reset an unused CVar that may be causing this.")) {
+                        CVarSetInteger("gAdvancedResolution.IntegerScale.ExceedBoundsBy", 0);
+                        CVarSave();
+                    }
+                }
+            } else {
+                UIWidgets::Spacer(enhancementSpacerHeight);
+            }
+
+            // I've ended up dummying this one out because it doesn't function in a satisfactory way.
+            // Consider this idea on the table, but I don't deem it an important enough feature to push for.
+            /*
+            UIWidgets::PaddedEnhancementCheckbox("Allow integer scale factor to go 1x above maximum screen bounds.",
+                                                 "gAdvancedResolution.IntegerScale.ExceedBoundsBy", false, false,
+                                                 !CVarGetInteger("gAdvancedResolution.PixelPerfectMode", 0), "",
+                                                 UIWidgets::CheckboxGraphics::Cross, false);
+            if (CVarGetInteger("gAdvancedResolution.IntegerScale.ExceedBoundsBy", 0)) {
+                ImGui::TextColored({ 0.0f, 0.85f, 0.85f, 1.0f },
+                                   " " ICON_FA_QUESTION_CIRCLE
+                                   " A scroll bar may become visible if screen bounds are exceeded.");
+            }*/
+
+        } // end of Additional Settings
+
+        // Clamp and update the CVars that don't use UIWidgets
+        if (IsBoolArrayTrue(update)) {
+            if (update[UPDATE_aspectRatioX]) {
+                if (aspectRatioX < 0.0f) {
+                    aspectRatioX = 0.0f;
+                }
+                CVarSetFloat("gAdvancedResolution.AspectRatioX", aspectRatioX);
+            }
+            if (update[UPDATE_aspectRatioY]) {
+                if (aspectRatioY < 0.0f) {
+                    aspectRatioY = 0.0f;
+                }
+                CVarSetFloat("gAdvancedResolution.AspectRatioY", aspectRatioY);
+            }
+            if (update[UPDATE_verticalPixelCount]) {
+                // There's a upper and lower clamp on the Libultraship side too,
+                // so clamping it here is purely visual, so the vertical resolution field reflects it.
+                if (verticalPixelCount < minVerticalPixelCount) {
+                    verticalPixelCount = minVerticalPixelCount;
+                }
+                if (verticalPixelCount > maxVerticalPixelCount) {
+                    verticalPixelCount = maxVerticalPixelCount;
+                }
+                CVarSetInteger("gAdvancedResolution.VerticalPixelCount", verticalPixelCount);
+            }
+            // Delay saving this set of CVars by a predetermined length of time, in frames.
+            updateCountdown = countdownStartingValue;
+        }
+        if (updateCountdown > 0) {
+            updateCountdown--;
+        } else {
+            CVarSave();
+        }
```

Comment:
This is ostensibly to prevent `CVarSave()` being called too often while typing in the number fields, but I don't know if this is all that necessary or not.